This commit combines a few simple changes.

adrianreber opened the pull request:
* Initialize the stats dict correctly.
* Store the time of the last crawl also if the crawl fails.
* Commit the database changes also if an error occured.

adrianreber commented:
This PR includes some simple one-line changes packed together.

ralphbean commented on utility/mm2_crawler:
```diff
@@ -1067,8 +1063,11 @@ def worker(options, config, host):
+    host.last_crawled = datetime.datetime.utcnow()
     logger.removeHandler(fh)
     fh.close()
+    session.commit()
```
Are you sure we want to commit even if an error occurred?  Typically, you don't want to do this as it might mean the database records are in some kind of inconsistent or unexpected (exceptional) state.

adrianreber commented:
Hmm... The reasoning behind this change was that the changes were only committed after a successful per_host() run. The changes were not committed for the case TimeoutException:.

As the last_time_crawled was not updated correctly in the mark_not_up2date() function I thought I move the commit() and last_time_crawled at the end so that it is done at one central point. Your are right, that it should probably not be committed if something failed.

So I will change this to only commit and write last_time_crawled in try: and TimeoutException:
The close can be left at the place I put it I think. I will update this PR with a new version.

adrianreber commented:
I have updated my branch and I am not only committing the changes if the crawl was successful or if it timed out. Not if it failed for another reason. I also moved the last_time_crawled just before the session.commit() in both cases session.close() is still after the exception handling.

ralphbean commented:
Looks good to me now.  :+1:!
